### ISSUE TITLE
Consolidate learnings from redux refactor

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#966]
## Description
Over the past several months, we have been migrating from using `react-query` to using an architecture with a `redux` store, a caching layer of "repositories" and "models" that handle most of the logic.

## Prerequisites
Don't start working on this until all is agreed.

## Requirements
These are the requirements that a finished refactor should accommodate.

- [ ] Consistent(?) way of handling mutations
- [ ] Support SSR

## Possible implementations
Unknown

## Open questions
Lots! To be discussed below.

**Comment by richardolsson** [comment]
Maybe one way of handling mutations could be to return special "futures" for mutations, and the model can store them until they resolve if it's necessary for UI feedback.

The repos could then return promises, and the model will return a future in a property, e.g.:

```ts
class MyModel {
  private _createPromise: Promise | null = null;
  // …

  get personBeingCreated(): IFuture | null {
    return this._createPromise && new PromiseFuture(_createPromise);
  }

  createPerson(data): IFuture {
    this._createPromise = this._repo.createPerson(data)
      .then(person => { this._createPromise = null });
    return new PromiseFuture(this._createPromise);
  }
}
```

**Comment by richardolsson** [comment]
This feels no longer relevant since the more recent refactor to rely on redux+hooks instead of redux+models. Future migrations to Next 14 will also mess with the SSR aspects of this. Closing as no longer relevant.
